include/picongpu/plugins/ChargeConservation.tpp

```diff
@@ -74,13 +68,11 @@ namespace picongpu
 
         Environment<>::get().PluginConnector().setNotificationPeriod(this, this->notifyPeriod);
 
-        pmacc::GridController<simDim>& con = pmacc::Environment<simDim>::get().GridController();
-        using namespace pmacc::math;
-        Size_t<simDim> gpuDim = (Size_t<simDim>) con.getGpuNodes();
-        zone::SphericZone<simDim> zone_allGPUs(gpuDim);
-        this->allGPU_reduce = AllGPU_reduce(new pmacc::algorithm::mpi::Reduce<simDim>(zone_allGPUs));
+        globalReduce = std::make_unique<algorithms::GlobalReduce>(1024);
```

Comment:
To me this looks like a magic number. Why do you call the constructor of `algorithms::GlobalReduce` with argument `1024`?

Reply:
Not magic, the class has since last week also a documentation ^^ https://github.com/ComputationalRadiationPhysics/picongpu/blob/12d0914e46046b41d8d93347f9ff933a601f626b/include/pmacc/algorithms/GlobalReduce.hpp#L42

Reply:
Never the less I changed it and use a local var now. IN this case using a local var is ok because the class not very seldom used and by using a local var with a proper name the reader is not required to check the doc string of the class^^ (Citat @psychocoderHPC : "Nobody is reading documentation!")